peer-to-peer-scheduler/peer-to-peer-scheduler.gs

```diff
@@ -0,0 +1,900 @@
+  }
+  showInfo(TITLE_SUCCESS, summary);
+}
+
+function cancelFullEvent(event) {
+  try {
+    // If the event was canceled previously (e.g. someone manually canceled), then deleteEvent() throws exception.
+    event.deleteEvent();
+    return true;
+  } catch (error) {
+    logBeginDelimiter();
+    console.log(ERROR_CODE_COULD_NOT_CANCEL_EVENT);
+    console.log(error);
+    logEndDelimiter();
+    return false;
+  }
+}
+
+function cancelEvents() {
+  let setupResult = initSetup();
+  if (!setupResult.success) {
+    return;
+  }
+  let { calendar, activeSheet, oneToOneData } = setupResult;
+  let calendarId = calendar.getId();
+
+  let canceledRows = [];
+  let invalidMeetings = [];
+  let partialCanceledRows = [];
+  let invalidDateTimeRows = [];
+  for (let rowIndex = 0; rowIndex < oneToOneData.length; ++rowIndex) {
+    let rowNumber = rowIndex + DATA_BEGIN_ROW;
+    let opsRequestFlag = parseInt(activeSheet.getRange(rowNumber, OPS_REQUEST_INDEX + 1).getValue());
+    if (opsRequestFlag !== FLAG_CANCEL_INVITE) {
+      // This row doesn't need to be canceled
+      continue;
+    }
+
+    let meetIdCell = activeSheet.getRange(rowNumber, MEETING_ID_INDEX + 1);
+    let meetLinkCell = activeSheet.getRange(rowNumber, MEETING_LINK_INDEX + 1);
+    let meetId = meetIdCell.getValue().trim();
+    if (meetId === "") {
+      // Meeting is not set
+      continue;
+    }
+
+    let event = getEvent(calendar, meetId);
+    if (!event) {
+      invalidMeetings.push(rowNumber);
+      continue;
+    }
+    
+    let cancelDatesCell = activeSheet.getRange(rowNumber, CANCELLATION_DATES_INDEX + 1);
+    let cancelDates = oneToOneData[rowIndex][CANCELLATION_DATES_INDEX].trim();
+    let startTime = parseTime(oneToOneData[rowIndex][START_TIME_INDEX].trim());
+    if (!startTime) {
+      invalidDateTimeRows.push(rowNumber);
+      continue;
+    }
+
+    if (cancelDates === "") { // Full Invite cancellation
+      let cancelStatus = cancelFullEvent(event);
+      if(!cancelStatus){
+        invalidMeetings.push(rowNumber);
+        continue;
+      }
+      canceledRows.push(rowNumber);
+      meetIdCell.clearContent();
+      meetLinkCell.clearContent();
+      continue;
+    }
+ 
+    let cancelDatesArray = cancelDates.split(",");
+    for (let i = 0; i < cancelDatesArray.length; ++i) {
+      let currentDate = parseDate(cancelDatesArray[i]);
+      if (!currentDate) {
+        invalidDateTimeRows.push(rowNumber);
+        continue;
+      }
+      cancelDatesArray[i] = new Date(currentDate.year, currentDate.month - 1, currentDate.day, startTime.hours, startTime.minutes);
+    }
+    
+    let instancesArray = getRecurringInstances(calendarId, meetId);
+    if(!instancesArray) {
+      invalidMeetings.push(rowNumber);
+      continue;
+    }
+    
+    let instancesDates = {} // For each instance, id format is "meetId_yyyymmddThhmmssZ" and start.dateTime format is "yyyy-mm-ddThh:mm:00+05:30"
+    instancesArray.forEach(
```

Comment:
Please use `for .. of` - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of

Reply:
Replaced `forEach` usage with `for...of`